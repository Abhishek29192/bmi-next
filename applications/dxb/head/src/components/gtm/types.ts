export type DataLayer = Record<string, unknown>[];

declare global {
  interface Window {
    dataLayer: DataLayer;
    [key: string]: DataLayer;
  }
}

export type GTMParams = {
  gtmIds: string[];
  dataLayer?: string[];
  dataLayerName?: string;
  auth?: string;
  preview?: string;
  includeInDevelopment?: boolean;
  enableWebVitalsTracking?: boolean;
  selfHostedOrigin?: string;
  defaultDataLayer?: { platform: string; env: string };
};

export type GAParams = {
  gaId: string;
  dataLayerName?: string;
};
