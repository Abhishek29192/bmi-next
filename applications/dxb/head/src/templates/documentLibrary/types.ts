import { Data as BreadcrumbsData } from "../../components/Breadcrumbs";
import { Data as PageData } from "../../components/Page";
import { Data as PageInfoData } from "../../components/PageInfo";
import { RichTextData } from "../../components/RichText";
import { Data as SiteData } from "../../components/Site";
import { DocumentsFilters } from "../../schema/resolvers/types/DocumentsFilters";
import { ContentfulDocument } from "../../types/Document";
import { ProductDocument } from "../../types/pim";
import { ResultType, Source, URLFilter } from "../../utils/filters";

export type QueryParams = {
  filters: URLFilter[];
};

export interface DocumentLibraryPageContext {
  pageId: string;
  siteId: string;
  categoryCode: string;
  variantCodeToPathMap?: Record<string, string>;
}

export interface ContentfulDocumentLibraryPage extends PageInfoData, PageData {
  description: RichTextData | null;
  allowFilterBy: string[] | null;
  source: Source;
  resultsType: ResultType;
  breadcrumbs: BreadcrumbsData;
  categoryCodes: string[];
  breadcrumbTitle: string;
  documentsFilters: DocumentsFilters | null;
}

export interface DocumentLibraryProps {
  pageContext: DocumentLibraryPageContext;
  data: {
    contentfulDocumentLibraryPage: ContentfulDocumentLibraryPage;
    contentfulSite: SiteData;
  };
}

export type DocumentType =
  | ContentfulDocument["__typename"]
  | ProductDocument["__typename"]
  | "";
