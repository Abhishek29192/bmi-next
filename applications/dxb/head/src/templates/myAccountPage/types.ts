import { Data as ContactDetailsData } from "../../components/ContactDetails";
import type { Data as BreadcrumbsData } from "../../components/Breadcrumbs";
import type { Data as SiteData } from "../../components/Site";
import type { HeroProps } from "./Hero";
import type { Data as LinkData } from "../../components/link/types";

export type ServiceSupportSectionProps = {
  titleForServiceSupportSection: string;
  serviceSupportCards: [ContactDetailsData, ...ContactDetailsData[]];
};

export type GlobalTools = "My profile" | "Trainings" | "Roof measurement";

export type LocalToolType =
  | "Calculator"
  | "Company Profile"
  | "Guarantees"
  | "Team"
  | "Visualiser";

export type LocalTool = {
  title: string;
  type: LocalToolType;
  link: LinkData;
};

export type ToolSectionProps = {
  titleForToolSection: string;
  globalTools: readonly [GlobalTools, ...GlobalTools[]];
  tools: readonly [LocalTool, ...LocalTool[]] | null;
  path: string;
};

export type AccountPage = {
  title: string;
  breadcrumbTitle: string | null;
  breadcrumbs: BreadcrumbsData;
} & ToolSectionProps &
  Omit<ServiceSupportSectionProps, "serviceSupportCards"> &
  HeroProps & {
    path: string;
    serviceSupportCards:
      | ServiceSupportSectionProps["serviceSupportCards"]
      | null;
  };

export type SiteDataWithAccountPage = Omit<SiteData, "accountPage"> & {
  accountPage: AccountPage;
};
