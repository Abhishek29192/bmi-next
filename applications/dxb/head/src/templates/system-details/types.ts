export type GalleryImageType = {
  mainSource: string;
  thumbnail: string;
  altText: string;
};

export type SystemProductImageType = SystemImageType & {
  format: string;
  containerId: string;
};

export type SystemImageType = {
  url: string;
  name: string;
  mime: string;
  fileSize: number;
  containerId: string;
  assetType: string;
  altText?: string;
  allowedToDownload: boolean;
  realFileName: string;
};

export type Category = {
  categoryType: string;
  name: string;
  image?: SystemImageType;
};

export type Feature = {
  code:
    | "bmiSystemsClassificationCatalog/1.0/systemAttributes.roofbuildup"
    | "bmiSystemsClassificationCatalog/1.0/systemAttributes.promotionalcontent";
  featureValues: { value: string }[];
  name: string;
};

export type Classification = {
  code: "systemAttributes" | "measurementAttributes";
  features: Feature[];
  name: string;
};

export interface SystemDetails {
  type?: string;
  approvalStatus?: string;
  code?: string;
  assets: any[];
  name: string;
  categories: Category[];
  classifications: Classification[];
  images: SystemProductImageType[] | null;
  longDescription?: string;
  shortDescription?: string;
  systemBenefits?: string[];
  systemLayers?: any[];
  systemReferences?: any[];
}
