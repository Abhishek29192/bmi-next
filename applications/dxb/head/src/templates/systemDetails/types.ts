import {
  Image,
  ClassificationFeatureUnit,
  Product
} from "../../components/types/ProductBaseTypes";

export type GalleryImageType = {
  mainSource: string;
  thumbnail: string;
  altText: string;
};

export type CategoryImage = {
  allowedToDownload: boolean;
  fileSize: number;
  mime: string;
  name: string;
  realFileName: string;
  url: string;
};

export type Category = {
  categoryType: string;
  name: string;
  image?: CategoryImage;
  code?: string;
};

export type FeatureValue = { value: string };

export type Feature = {
  code: string;
  featureValues: FeatureValue[];
  name: string;
  featureUnit?: ClassificationFeatureUnit;
};

export type Classification = {
  code:
    | "systemAttributes"
    | "measurementAttributes"
    | "scoringWeightAttributes"
    | "ScoringWeightAttributes"
    | "uniquesellingpropositionsAttributes";
  features: Feature[];
  name: string;
};

export type AssetType =
  | "BIM"
  | "WARRANTIES"
  | "AWARDS"
  | "GUARANTIES"
  | "CERTIFICATES"
  | "SPECIFICATION"
  | "CAD"
  | "DATA_SHEETS";

export type MimeType =
  | "image/jpeg"
  | "image/png"
  | "image/tiff"
  | "application/pdf";

export type Assets = {
  allowedToDownload: boolean;
  assetType: AssetType;
  fileSize?: number;
  mime: MimeType;
  name: string;
  realFileName: string; // includes file extension
  url: string;
  format?: string;
};

export type SystemBenefits = string[];

interface ProductCode {
  code: string;
}

export type SystemLayer = {
  addon: boolean;
  approvalStatus: "approved" | "unapproved";
  code: string;
  images: Image[];
  layerNumber: number;
  longDescription: string;
  name: string;
  optionalProducts: ProductCode[];
  products: ProductCode[];
  shortDescription: string;
  type: string;
  relatedOptionalProducts: Product[];
  relatedProducts: Product[];
};

export interface SystemDetails {
  type?: string;
  approvalStatus?: string;
  code?: string;
  assets: Assets[];
  name: string;
  categories: Category[];
  classifications: Classification[];
  images: Image[] | null;
  longDescription: string;
  shortDescription?: string;
  systemBenefits: SystemBenefits;
  systemLayers?: SystemLayer[];
  systemReferences?: any[];
}
