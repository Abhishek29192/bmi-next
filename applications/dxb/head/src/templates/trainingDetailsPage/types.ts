import type { Course } from "@bmi/gatsby-source-docebo";
import type { Data as BreadcrumbsData } from "../../components/Breadcrumbs";

export type Session = {
  id: number;
  code: string | null;
  name: string;
  date_start: number;
  date_end: number;
};

export type TrainingDetailsCourseType = Pick<
  Course,
  | "code"
  | "id_course"
  | "slug_name"
  | "name"
  | "course_type"
  | "categoryName"
  | "description"
  | "img_url"
  | "price"
  | "currencySymbol"
> & {
  breadcrumbs: BreadcrumbsData;
  sessions: Session[];
};
