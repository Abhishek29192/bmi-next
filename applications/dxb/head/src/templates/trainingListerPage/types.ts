import { Training } from "@bmi/elasticsearch-types";
import { Data as PageData } from "../../components/Page";
import { Data as SiteData } from "../../components/Site";
import { Data as ImageData } from "../../components/Image";
import { Data as TitleWithContentData } from "../../components/TitleWithContent";

export type EsCollapsedTraining = {
  inner_hits: {
    inner_hits: {
      hits: {
        total: {
          value: number;
          relation: string;
        };
        max_score: number;
        hits: {
          _index: string;
          _type: "_doc";
          _id: string;
          _score: number;
          _source: Training;
        }[];
      };
    };
  };
  _source: Training;
};

export type EsTrainingHit = {
  _index: string;
  _type: "_doc";
  _id: string;
  _score: number;
  _source: Training;
};

export type EsResponse<T extends EsCollapsedTraining | EsTrainingHit> = {
  hits: {
    total: {
      value: number;
      relation: string;
    };
    max_score: number | null;
    hits: T[];
  };
};

export type TrainingListerPageData = Omit<PageData, "signupBlock"> & {
  __typename: "ContentfulTrainingListerPage";
  title: string;
  subtitle: string | null;
  breadcrumbTitle: string | null;
  featuredMedia: ImageData;
  searchTips: TitleWithContentData;
};

export type TrainingListerPageProps = {
  data: {
    contentfulTrainingListerPage: TrainingListerPageData;
    contentfulSite: SiteData;
  };
  pageContext: {
    variantCodeToPathMap?: Record<string, string>;
  };
};
