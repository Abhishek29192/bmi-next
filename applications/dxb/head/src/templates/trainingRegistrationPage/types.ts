import type { Training } from "@bmi/elasticsearch-types";
import type { ESResponse } from "../../types/elasticsearch";
import type { Data as SiteData } from "../../components/Site";
import type { Data as BreadcrumbsData } from "../../components/Breadcrumbs";

export type TrainingRegistrationPageData = {
  __typename: "ContentfulTrainingRegistrationPage";
  breadcrumbs: BreadcrumbsData;
  breadcrumbTitle: string | null;
  path: string;
  title: string;
  salutationTitle: string;
  salutationMale: string;
  salutationFemale: string;
  firstName: string;
  lastName: string;
  email: string;
  companyName: string;
  position: string;
  customerNumber: string;
  city: string;
  street: string;
  postalCode: string;
  phoneNumber: string;
  competentChamberLabel: {
    competentChamberLabel: string;
  };
  competentChamber: string;
  bmiSystemPartnerClubTitle: string;
  isMemberOfBmiLabel: string;
  isNotMemberOfBmiLabel: string;
  discoverySourceTitle: string;
  discoverySourceBrochure: string;
  discoverySourceFieldService: string;
  discoverySourceWebsite: string;
  discoverySourceFacebook: string;
  discoverySourceInstagram: string;
  discoverySourceXing: string;
  discoverySourceLinkedin: string;
  discoverySourceOther: string;
  discoverySourceSpecifyOther: string;
  comment: string;
  consentText: {
    consentText: string;
  };
  termsOfUse: {
    termsOfUse: string;
  };
  registerButton: string;
};

export type TrainingRegistrationPageProps = {
  data: {
    contentfulTrainingRegistrationPage: TrainingRegistrationPageData;
    contentfulTrainingListerPage: {
      path: string;
      breadcrumbs: BreadcrumbsData;
    } | null;
    contentfulSite: SiteData;
  };
};

export type ESTrainingDetails = ESResponse<Training>;
