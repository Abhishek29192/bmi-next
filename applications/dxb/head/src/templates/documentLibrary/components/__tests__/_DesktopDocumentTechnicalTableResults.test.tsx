import { render } from "@testing-library/react";
import React from "react";
import fileIconsMap from "../../../../components/FileIconsMap";
import { ContentfulAssetType as AssetTypeData } from "../../../../types/AssetType";
import { ProductDocument } from "../../../../types/pim";
import createAssetType from "../../../../__tests__/helpers/AssetTypeHelper";
import createPimDocument from "../../../../__tests__/helpers/PimDocumentHelper";
import DesktopDocumentTechnicalTableResults from "../_DesktopDocumentTechnicalTableResults";

describe("DesktopDocumentTechnicalTableResults component", () => {
  describe("Renders correctly", () => {
    it("when only single documents are present for asset types", () => {
      const assetTypes: AssetTypeData[] = [createAssetType()];
      const documentsByProduct: [string, ProductDocument[]][] = [
        ["product1", [createPimDocument()]],
        ["product3", []]
      ];

      const wrapper = render(
        <DesktopDocumentTechnicalTableResults
          documentsByProduct={documentsByProduct}
          assetTypes={assetTypes}
          fileIconsMap={fileIconsMap}
        />
      );

      expect(wrapper.baseElement).toMatchSnapshot();
    });
    it("when not all products have all asset types", () => {
      const assetTypes: AssetTypeData[] = [
        createAssetType(),
        createAssetType({
          id: "a different asset ID",
          name: "doesn't matter but looks good"
        })
      ];
      const documentsByProduct: [string, ProductDocument[]][] = [
        ["product1", [createPimDocument({ assetType: assetTypes[0] })]],
        ["product3", []]
      ];

      const wrapper = render(
        <DesktopDocumentTechnicalTableResults
          documentsByProduct={documentsByProduct}
          assetTypes={assetTypes}
          fileIconsMap={fileIconsMap}
        />
      );

      expect(wrapper.baseElement).toMatchSnapshot();
    });

    it("when multiple documents are present for asset types", () => {
      const assetTypes: AssetTypeData[] = [createAssetType()];
      const documentsByProduct: [string, ProductDocument[]][] = [
        ["product1", [createPimDocument(), createPimDocument()]],
        ["product3", []]
      ];

      const wrapper = render(
        <DesktopDocumentTechnicalTableResults
          documentsByProduct={documentsByProduct}
          assetTypes={assetTypes}
          fileIconsMap={fileIconsMap}
        />
      );

      expect(wrapper.baseElement).toMatchSnapshot();
    });
  });
});
