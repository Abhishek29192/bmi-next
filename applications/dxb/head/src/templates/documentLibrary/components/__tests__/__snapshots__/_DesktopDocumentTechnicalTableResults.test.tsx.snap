// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DesktopDocumentTechnicalTableResults component Renders correctly when multiple documents are present for asset types 1`] = `
<body>
  <div>
    <div
      class="table-div"
    >
      <div>
        <table
          class="MuiTable-root Table Table--row-no-color"
        >
          <thead
            class="MuiTableHead-root ColorPair ColorPair--pearl"
          >
            <tr
              class="MuiTableRow-root header-row MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                MC: documentLibrary.headers.product
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head asset-type-cell"
                scope="col"
              >
                asset-code
                <div
                  class="Tooltip"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root iconButtonStyles-root-60 iconButtonStyles-42-59 iconButtonStyles-text-62"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root tooltip-icon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head all-files-header"
                scope="col"
              >
                <span
                  class="all-files-header-wrapper"
                >
                  <svg
                    class="all-files-icon"
                  />
                  <span>
                    MC: documentLibrary.headers.allFiles
                  </span>
                </span>
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root row"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                product name
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body align-center"
              >
                <svg
                  class="no-document-icon"
                />
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body align-center"
              >
                <span
                  class=""
                >
                  <div
                    class="MuiFormControl-root Checkbox"
                  >
                    <label
                      class="MuiFormControlLabel-root"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-69 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <input
                            aria-label="MC: documentLibrary.download product1"
                            class="PrivateSwitchBase-input-72"
                            data-indeterminate="false"
                            name="product-base-code-0"
                            type="checkbox"
                            value=""
                          />
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                      />
                    </label>
                  </div>
                </span>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root row"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                product3
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body align-center"
              >
                <svg
                  class="no-document-icon"
                />
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body align-center"
              />
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</body>
`;

exports[`DesktopDocumentTechnicalTableResults component Renders correctly when not all products have all asset types 1`] = `
<body>
  <div>
    <div
      class="table-div"
    >
      <div>
        <table
          class="MuiTable-root Table Table--row-no-color"
        >
          <thead
            class="MuiTableHead-root ColorPair ColorPair--pearl"
          >
            <tr
              class="MuiTableRow-root header-row MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                MC: documentLibrary.headers.product
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head asset-type-cell"
                scope="col"
              >
                asset-code
                <div
                  class="Tooltip"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root iconButtonStyles-root-36 iconButtonStyles-42-35 iconButtonStyles-text-38"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root tooltip-icon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head asset-type-cell"
                scope="col"
              >
                asset-code
                <div
                  class="Tooltip"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root iconButtonStyles-root-36 iconButtonStyles-42-35 iconButtonStyles-text-38"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root tooltip-icon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head all-files-header"
                scope="col"
              >
                <span
                  class="all-files-header-wrapper"
                >
                  <svg
                    class="all-files-icon"
                  />
                  <span>
                    MC: documentLibrary.headers.allFiles
                  </span>
                </span>
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root row"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                product name
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body align-center"
              >
                <a
                  data-gtm="{\\"id\\":\\"download1\\",\\"label\\":\\"Download\\",\\"action\\":\\"http://localhost:9000/document.pdf\\"}"
                  download="pim document title"
                  href="http://localhost:9000/document.pdf"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <svg
                    class="format-icon"
                  />
                </a>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body align-center"
              >
                <a
                  data-gtm="{\\"id\\":\\"download1\\",\\"label\\":\\"Download\\",\\"action\\":\\"http://localhost:9000/document.pdf\\"}"
                  download="pim document title"
                  href="http://localhost:9000/document.pdf"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <svg
                    class="format-icon"
                  />
                </a>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body align-center"
              >
                <span
                  class=""
                >
                  <div
                    class="MuiFormControl-root Checkbox"
                  >
                    <label
                      class="MuiFormControlLabel-root"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-45 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <input
                            aria-label="MC: documentLibrary.download product1"
                            class="PrivateSwitchBase-input-48"
                            data-indeterminate="false"
                            name="product-base-code-0"
                            type="checkbox"
                            value=""
                          />
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                      />
                    </label>
                  </div>
                </span>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root row"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                product3
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body align-center"
              >
                <svg
                  class="no-document-icon"
                />
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body align-center"
              >
                <svg
                  class="no-document-icon"
                />
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body align-center"
              />
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</body>
`;

exports[`DesktopDocumentTechnicalTableResults component Renders correctly when only single documents are present for asset types 1`] = `
<body>
  <div>
    <div
      class="table-div"
    >
      <div>
        <table
          class="MuiTable-root Table Table--row-no-color"
        >
          <thead
            class="MuiTableHead-root ColorPair ColorPair--pearl"
          >
            <tr
              class="MuiTableRow-root header-row MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                MC: documentLibrary.headers.product
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head asset-type-cell"
                scope="col"
              >
                asset-code
                <div
                  class="Tooltip"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root iconButtonStyles-root-12 iconButtonStyles-42-11 iconButtonStyles-text-14"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root tooltip-icon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head all-files-header"
                scope="col"
              >
                <span
                  class="all-files-header-wrapper"
                >
                  <svg
                    class="all-files-icon"
                  />
                  <span>
                    MC: documentLibrary.headers.allFiles
                  </span>
                </span>
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root row"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                product name
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body align-center"
              >
                <svg
                  class="no-document-icon"
                />
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body align-center"
              >
                <span
                  class=""
                >
                  <div
                    class="MuiFormControl-root Checkbox"
                  >
                    <label
                      class="MuiFormControlLabel-root"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-21 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <input
                            aria-label="MC: documentLibrary.download product1"
                            class="PrivateSwitchBase-input-24"
                            data-indeterminate="false"
                            name="product-base-code-0"
                            type="checkbox"
                            value=""
                          />
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                      />
                    </label>
                  </div>
                </span>
              </td>
            </tr>
            <tr
              class="MuiTableRow-root row"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                product3
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body align-center"
              >
                <svg
                  class="no-document-icon"
                />
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body align-center"
              />
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</body>
`;
